test/k8sT/Services.go

```diff
@@ -1221,6 +1221,42 @@ var _ = Describe("K8sServicesTest", func() {
 			testNodePort(false, false, false, 0)
 		})
 
+		testMaglev := func() {
+			var (
+				data  v1.Service
+				count = 10
+			)
+
+			err := kubectl.Get(helpers.DefaultNamespace, "service echo").Unmarshal(&data)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot retrieve service")
+
+			for _, port := range []int{60000, 61000, 62000} {
+				dstPod := ""
+
+				// Send requests from the same IP and port to different nodes, and check
+				// that the same backend is selected
+
+				for _, host := range []string{k8s1IP, k8s2IP} {
+					url := getTFTPLink(host, data.Spec.Ports[1].NodePort)
+					cmd := helpers.CurlFail("--local-port %d %s", port, url) + " | grep 'Hostname:' " // pod name is in the hostname
+
+					By("Making %d HTTP requests from %s:%d to %q", count, outsideNodeName, port, url)
+
+					for i := 1; i <= count; i++ {
+						res := kubectl.ExecInHostNetNS(context.TODO(), outsideNodeName, cmd)
```

Comment:
We've had problems in the past with looping to establish connections into pods tens of times in a row to validate a series of LB decisions. Is this following the latest versions of the equivalent code? I was under the impression that these days we assemble a single command to run several checks in a small number of `kubectl exec ...` attempts to mitigate flakes due to this. Maybe ping @jrajahalme if this doesn't sound familiar.

Reply:
I'm familiar with the loop "unrolling" feature, as I've introduced it - 58be565cfc418461ede946fa7270edf957db0187 :-) Unfortunately, we cannot use it here, because we need to check for `Hostname` return value. Extending the existing check https://github.com/cilium/cilium/blob/master/test/k8sT/Services.go#L155 would make it more complex and less readable.

Created https://github.com/cilium/cilium/issues/13149 to suggest an alternative approach which would allow to "unroll" the exec's.

Reply:
haha, OK awesome, thanks for the explanation.